qutip/tests/core/data/test_mathematics.py

```diff
@@ -770,3 +771,22 @@ def op_numpy(self, matrix):
         pytest.param(data.transpose_csr, CSR, CSR),
         pytest.param(data.transpose_dense, Dense, Dense),
     ]
+
+
+class TestProject(UnaryOpMixin):
+    def op_numpy(self, matrix):
+        if matrix.shape[0] == 1:
+            return np.outer(np.conj(matrix), matrix)
+        else:
+            return np.outer(matrix, np.conj(matrix))
+
+    shapes = [
+        (pytest.param((1, 1), id="scalar"),),
+        (pytest.param((1, 100), id="bra"),),
+        (pytest.param((100, 1), id="ket"),),
+    ]
+
```

Comment:
If I remember correctly, you should also supply some `bad_shapes` to test as well, making sure that you can project an operator?